subprojects/core/src/main/java/org/gradle/internal/classpath/Instrumented.java

```diff
@@ -142,8 +143,13 @@ private static List<CallInterceptor> getGeneratedCallInterceptorsFor(String clas
         try {
             Class<?> generatedClass = Class.forName(className);
             Method callInterceptorsGetter = generatedClass.getDeclaredMethod("getCallInterceptors");
+            // Generated classes are not public and in the org.gradle.internal.classpath.generated package
+            callInterceptorsGetter.setAccessible(true);
```

Comment:
Would making the generated classes public turn out cleaner? Also, I would like to reference them from tests at some point.

Reply:
Yes, that will make it cleaner. So I can just make them public. And if we need to forbid package access in the future, we can move that method to some `org.gradle.internal.classpath.generated.InterceptorFactory` class

Reply:
Classes are now public